android/engine/src/main/java/org/smartregister/fhircore/engine/ui/pin/PinSetupScreen.kt

```diff
@@ -0,0 +1,199 @@
+import androidx.compose.foundation.layout.requiredHeight
+import androidx.compose.foundation.layout.requiredWidth
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.wrapContentWidth
+import androidx.compose.material.Button
+import androidx.compose.material.DropdownMenu
+import androidx.compose.material.DropdownMenuItem
+import androidx.compose.material.Icon
+import androidx.compose.material.IconButton
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.material.TopAppBar
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.ArrowBack
+import androidx.compose.material.icons.outlined.MoreVert
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.livedata.observeAsState
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import org.smartregister.fhircore.engine.R
+import org.smartregister.fhircore.engine.ui.components.PinView
+import org.smartregister.fhircore.engine.ui.login.APP_LOGO_TAG
+import org.smartregister.fhircore.engine.util.annotation.ExcludeFromJacocoGeneratedReport
+
+@Composable
+fun PinSetupScreen(viewModel: PinViewModel) {
+
+  val inputPin by viewModel.pin.observeAsState(initial = "")
+  val enableSetPin by viewModel.enableSetPin.observeAsState(initial = false)
+
+  PinSetupPage(
+    onPinChanged = viewModel::onPinChanged,
+    inputPin = inputPin,
+    setPinEnabled = enableSetPin ?: false,
+    onPinConfirmed = viewModel::onPinConfirmed,
+    onMenuSettingClicked = { viewModel.onMenuSettingClicked() },
+  )
+}
+
+@OptIn(ExperimentalComposeUiApi::class)
+@Composable
+fun PinSetupPage(
+  modifier: Modifier = Modifier,
+  onPinChanged: (String) -> Unit,
+  inputPin: String,
+  setPinEnabled: Boolean = false,
+  onPinConfirmed: () -> Unit,
+  onMenuSettingClicked: () -> Unit,
+) {
+
+  var showMenu by remember { mutableStateOf(false) }
+
+  Surface(color = colorResource(id = R.color.white_slightly_opaque)) {
+    TopAppBar(
+      title = { Text(text = "", Modifier.testTag(PIN_TOOLBAR_TITLE)) },
+      navigationIcon = {
+        IconButton(onClick = {}) {
+          Icon(
+            Icons.Filled.ArrowBack,
+            contentDescription = "Back arrow",
```

Comment:
this text is for view reference only, it will not be reflecting in application and doesn't required MLS.
so no need to add extra string in string.xml